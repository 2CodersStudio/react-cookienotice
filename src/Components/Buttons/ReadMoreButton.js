// @flow

import * as React from 'react'
import styled from 'styled-components'
import clsx from 'clsx'

const ButtonStyled = styled.a`
  font-size: 16px;
  height: auto;
  color: #b5b5b5;
  padding: 10px 18px;
  border: 1px solid #b5b5b5;
  border-radius: 24px;
  box-shadow: none;
  text-decoration: none;
  white-space: nowrap;
  transition: all 0.5s ease 0s;

  &:hover {
    color: #000000;
    border-color: #000000;
  }

  &:not(:first-child) {
    margin-left: 8px;
  }

  &.dark {
    &:hover {
      color: #ffffff;
      border-color: #ffffff;
    }
  }

  @media (max-width: 768px) {
    &:not(:first-child) {
      margin-left: 0;
    }

    font-size: 12px;
    margin: 5px 0;
  }
`

type Props = {|
  label: string,
  link: string,
  openInNewTab: boolean,
  darkTheme: boolean,
|}

export default class ReadMoreButton extends React.Component<Props> {
  render(): React.Node {
    const { label, link, openInNewTab, darkTheme } = this.props

    return (
      <ButtonStyled
        className={clsx('react-cookienotice-read-more-button', {
          dark: darkTheme,
        })}
        href={link}
        target={openInNewTab ? '_blank' : '_self'}
      >
        {label}
      </ButtonStyled>
    )
  }
}
