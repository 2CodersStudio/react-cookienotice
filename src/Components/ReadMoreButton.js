// @flow

import React from 'react'
import styled from 'styled-components'

const Button = styled.a`
  font-size: 16px;
  height: auto;
  color: #8c97a1;
  padding: 10px 18px;
  border: 1px solid #e0e2e4;
  border-radius: 24px;
  box-shadow: none;
  text-decoration: none;
  white-space: nowrap;
  transition: all 0.5s ease 0s;

  &:hover {
    color: #1d1148;
    border-color: #1d1148;
    background: #ffffff;
  }

  &:not(:first-child) {
    margin-left: 8px;
  }

  @media (max-width: 768px) {
    font-size: 14px;
  }
`

type Props = {|
  label?: string,
  link?: string,
  openInNewTab?: boolean,
|}

export default class ReadMoreButton extends React.Component<Props> {
  computeOpenInNewTab(): '_blank' | '_self' {
    const { openInNewTab } = this.props

    if (typeof openInNewTab === 'undefined') {
      return '_blank'
    }

    return openInNewTab ? '_blank' : '_self'
  }

  render() {
    const { label, link } = this.props

    return (
      <Button
        className='read-more-button'
        href={link || 'http://aboutcookies.org/'}
        target={this.computeOpenInNewTab()}
      >
        {label || 'Read more'}
      </Button>
    )
  }
}
