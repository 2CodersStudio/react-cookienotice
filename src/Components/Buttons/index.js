// @flow

import * as React from 'react'
import styled from 'styled-components'
import AcceptButton from './AcceptButton'
import ReadMoreButton from './ReadMoreButton'

const ButtonsWrapper = styled.div`
  display: flex;
  align-items: center;

  @media (max-width: 768px) {
    flex-direction: column;
  }
`

type Props = {|
  acceptButtonLabel: string,
  onAcceptButtonClick: () => void,
  readMoreButtonLabel: string,
  readMoreButtonLink: string,
  readMoreButtonOpenInNewTab: boolean,
  reverseButtons: boolean,
|}

export default class Buttons extends React.Component<Props> {
  render(): React.Node {
    const {
      acceptButtonLabel,
      onAcceptButtonClick,
      readMoreButtonLabel,
      readMoreButtonLink,
      readMoreButtonOpenInNewTab,
      reverseButtons,
    } = this.props

    const buttons = [
      <AcceptButton
        key='accept-button'
        label={acceptButtonLabel}
        onButtonClick={onAcceptButtonClick}
      />,
      <ReadMoreButton
        key='read-more-button'
        label={readMoreButtonLabel}
        link={readMoreButtonLink}
        openInNewTab={readMoreButtonOpenInNewTab}
      />,
    ]

    return (
      <ButtonsWrapper className='buttons-wrapper'>
        {reverseButtons ? buttons.reverse() : buttons}
      </ButtonsWrapper>
    )
  }
}
