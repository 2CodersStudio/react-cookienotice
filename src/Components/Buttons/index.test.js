// @flow

import * as React from 'react'
import { shallow } from 'enzyme'
import Buttons from './index'
import AcceptButton from './AcceptButton'
import ReadMoreButton from './ReadMoreButton'

describe('Buttons', () => {
  it('should render', () => {
    const wrapper = (
      <Buttons
        acceptButtonLabel='acceptButtonLabel'
        onAcceptButtonClick={() => {}}
        readMoreButtonLabel='readMoreButtonLabel'
        readMoreButtonLink='readMoreButtonLink'
        readMoreButtonOpenInNewTab={false}
        reverseButtons={false}
        darkTheme={false}
      />
    )

    expect(wrapper).toMatchSnapshot()
  })

  describe('should manage reverse buttons', () => {
    it('without reverse', () => {
      const wrapper = shallow(
        <Buttons
          acceptButtonLabel='acceptButtonLabel'
          onAcceptButtonClick={() => {}}
          readMoreButtonLabel='readMoreButtonLabel'
          readMoreButtonLink='readMoreButtonLink'
          readMoreButtonOpenInNewTab={false}
          reverseButtons={false}
          darkTheme={false}
        />
      )

      expect(wrapper.childAt(0).is(AcceptButton)).toBeTruthy()
      expect(wrapper.childAt(1).is(ReadMoreButton)).toBeTruthy()
    })

    it('with reverse', () => {
      const wrapper = shallow(
        <Buttons
          acceptButtonLabel='acceptButtonLabel'
          onAcceptButtonClick={() => {}}
          readMoreButtonLabel='readMoreButtonLabel'
          readMoreButtonLink='readMoreButtonLink'
          readMoreButtonOpenInNewTab={false}
          reverseButtons
          darkTheme={false}
        />
      )

      expect(wrapper.childAt(0).is(ReadMoreButton)).toBeTruthy()
      expect(wrapper.childAt(1).is(AcceptButton)).toBeTruthy()
    })
  })
})
