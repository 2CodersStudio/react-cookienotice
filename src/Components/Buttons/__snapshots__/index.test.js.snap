// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Buttons should render 1`] = `
<Buttons
  acceptButtonLabel="acceptButtonLabel"
  darkTheme={false}
  onAcceptButtonClick={[Function]}
  readMoreButtonLabel="readMoreButtonLabel"
  readMoreButtonLink="readMoreButtonLink"
  readMoreButtonOpenInNewTab={false}
  reverseButtons={false}
/>
`;
